### DISCUSSION TITLE
The server responded with status 404 (Faraday::ResourceNotFound)

### DISCUSSION THREAD

**Discussion by thpthang-nextfunc** [fastlane/fastlane#20846] [Q&A]
fastlane deploy
Success 1st time, but 2nd time it failed.

```
<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                                          |
| --------------------------- | -------------------------------------------------------------- |
| OS                          | 12.6.1                                                         |
| Ruby                        | 2.6.10                                                         |
| Bundler?                    | false                                                          |
| Git                         | git version 2.32.1 (Apple Git-133)                             |
| Installation Source         | /usr/local/bin/fastlane                                        |
| Host                        | macOS 12.6.1 (21G217)                                          |
| Ruby Lib Dir                | /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib |
| OpenSSL Version             | LibreSSL 2.8.3                                                 |
| Is contained                | false                                                          |
| Is homebrew                 | false                                                          |
| Is installed via Fabric.app | false                                                          |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/                    |
| Xcode Version               | 13.4                                                           |
| Swift Version               | 5.6.1                                                          |


### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |


### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
# This file contains the fastlane.tools configuration
# You can find the documentation at https://docs.fastlane.tools
#
# For a list of all available actions, check out
#
#     https://docs.fastlane.tools/actions
#
# For a list of all available plugins, check out
#
#     https://docs.fastlane.tools/plugins/available-plugins
#

# Uncomment the line if you want fastlane to automatically update itself
# update_fastlane

default_platform(:android)

platform :android do
  desc "Runs all the tests"
  lane :deploy do
    begin
    # 
    firebase_app_distribution(
      # groups: "Test team",
      release_notes: "My release notes",
      apk_path: "../build/app/outputs/flutter-apk/app-release.apk",
      firebase_cli_path: "/usr/local/bin/firebase",
      firebase_cli_token: "...",
      app: "...",
    )
    # 
    end
  end
end

```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
json_key_file("") # Path to the json secret file - Follow https://docs.fastlane.tools/actions/supply/#setup to get one
package_name("com.bumbii.app") # e.g. com.krausefx.app

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.210.1 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

| Plugin                                    | Version | Update-Status |
| ----------------------------------------- | ------- | ------------- |
| fastlane-plugin-firebase_app_distribution | 0.3.7   | ✅ Up-To-Date  |

<details><summary><b>Loaded gems</b></summary>

| Gem                                       | Version      |
| ----------------------------------------- | ------------ |
| did_you_mean                              | 1.3.0        |
| atomos                                    | 0.1.3        |
| claide                                    | 1.1.0        |
| colored2                                  | 3.1.2        |
| nanaimo                                   | 0.3.0        |
| rexml                                     | 3.2.5        |
| xcodeproj                                 | 1.22.0       |
| rouge                                     | 2.0.7        |
| xcpretty                                  | 0.3.0        |
| terminal-notifier                         | 2.0.0        |
| unicode-display_width                     | 1.8.0        |
| terminal-table                            | 1.8.0        |
| plist                                     | 3.6.0        |
| addressable                               | 2.8.1        |
| multipart-post                            | 2.0.0        |
| word_wrap                                 | 1.0.0        |
| optparse                                  | 0.1.1        |
| tty-screen                                | 0.8.1        |
| tty-cursor                                | 0.7.1        |
| tty-spinner                               | 0.9.3        |
| artifactory                               | 3.0.15       |
| babosa                                    | 1.0.4        |
| colored                                   | 1.2          |
| highline                                  | 2.0.3        |
| commander                                 | 4.6.0        |
| faraday-em_http                           | 1.0.0        |
| faraday-em_synchrony                      | 1.0.0        |
| faraday-excon                             | 1.1.0        |
| faraday-httpclient                        | 1.0.1        |
| faraday-multipart                         | 1.0.4        |
| faraday-net_http                          | 1.0.1        |
| faraday-net_http_persistent               | 1.2.0        |
| faraday-patron                            | 1.0.0        |
| faraday-rack                              | 1.0.0        |
| faraday-retry                             | 1.0.3        |
| ruby2_keywords                            | 0.0.5        |
| faraday                                   | 1.10.2       |
| unf_ext                                   | 0.0.8.2      |
| unf                                       | 0.1.4        |
| domain_name                               | 0.5.20190701 |
| http-cookie                               | 1.0.5        |
| faraday-cookie_jar                        | 0.0.7        |
| faraday_middleware                        | 1.2.0        |
| fastimage                                 | 2.2.6        |
| gh_inspector                              | 1.1.3        |
| mini_magick                               | 4.11.0       |
| naturally                                 | 2.2.1        |
| rubyzip                                   | 2.3.2        |
| security                                  | 0.1.3        |
| xcpretty-travis-formatter                 | 1.0.1        |
| dotenv                                    | 2.8.1        |
| bundler                                   | 1.17.2       |
| simctl                                    | 1.6.8        |
| jwt                                       | 2.5.0        |
| uber                                      | 0.1.0        |
| declarative                               | 0.0.20       |
| trailblazer-option                        | 0.1.2        |
| representable                             | 3.2.0        |
| retriable                                 | 3.1.2        |
| mini_mime                                 | 1.1.2        |
| memoist                                   | 0.16.2       |
| multi_json                                | 1.15.0       |
| os                                        | 1.1.4        |
| signet                                    | 0.17.0       |
| googleauth                                | 1.3.0        |
| httpclient                                | 2.8.3        |
| google-apis-core                          | 0.9.1        |
| google-apis-playcustomapp_v1              | 0.12.0       |
| google-cloud-env                          | 1.6.0        |
| google-cloud-errors                       | 1.3.0        |
| google-cloud-core                         | 1.6.0        |
| google-apis-iamcredentials_v1             | 0.16.0       |
| google-apis-storage_v1                    | 0.19.0       |
| digest-crc                                | 0.6.4        |
| google-cloud-storage                      | 1.44.0       |
| emoji_regex                               | 3.2.3        |
| aws-eventstream                           | 1.2.0        |
| aws-sigv4                                 | 1.5.2        |
| aws-sdk-kms                               | 1.59.0       |
| aws-sdk-s3                                | 1.117.1      |
| CFPropertyList                            | 3.0.5        |
| public_suffix                             | 5.0.0        |
| excon                                     | 0.94.0       |
| json                                      | 2.6.2        |
| webrick                                   | 1.7.0        |
| google-apis-androidpublisher_v3           | 0.31.0       |
| rake                                      | 13.0.6       |
| aws-partitions                            | 1.659.0      |
| jmespath                                  | 1.6.1        |
| aws-sdk-core                              | 3.167.0      |
| forwardable                               | 1.2.0        |
| logger                                    | 1.3.0        |
| date                                      | 2.0.3        |
| stringio                                  | 0.0.2        |
| ipaddr                                    | 1.2.2        |
| openssl                                   | 2.1.2        |
| zlib                                      | 1.0.0        |
| mutex_m                                   | 0.1.0        |
| ostruct                                   | 0.1.0        |
| strscan                                   | 1.0.0        |
| io-console                                | 0.4.7        |
| fileutils                                 | 1.1.0        |
| etc                                       | 1.0.1        |
| libxml-ruby                               | 3.2.1        |
| psych                                     | 3.1.0        |
| fastlane-plugin-firebase_app_distribution | 0.3.7        |
</details>

*generated on:* **2022-11-11**
</details>
```

**Comment by thpthang-nextfunc** [comment]
```
You can ⌘ + double-click on links to open them directly in your browser.
/Library/Ruby/Gems/2.6.0/gems/faraday-1.10.2/lib/faraday/response/raise_error.rb:22:in `on_complete': [!] the server responded with status 404 (Faraday::ResourceNotFound)
        from /Library/Ruby/Gems/2.6.0/gems/faraday-1.10.2/lib/faraday/middleware.rb:19:in `block in call'
        from /Library/Ruby/Gems/2.6.0/gems/faraday-1.10.2/lib/faraday/response.rb:61:in `on_complete'
        from /Library/Ruby/Gems/2.6.0/gems/faraday-1.10.2/lib/faraday/middleware.rb:18:in `call'
        from /Library/Ruby/Gems/2.6.0/gems/faraday_middleware-1.2.0/lib/faraday_middleware/response_middleware.rb:36:in `call'
        from /Library/Ruby/Gems/2.6.0/gems/faraday-1.10.2/lib/faraday/rack_builder.rb:154:in `build_response'
        from /Library/Ruby/Gems/2.6.0/gems/faraday-1.10.2/lib/faraday/connection.rb:516:in `run_request'
        from /Library/Ruby/Gems/2.6.0/gems/faraday-1.10.2/lib/faraday/connection.rb:283:in `post'
        from /Library/Ruby/Gems/2.6.0/gems/fastlane-plugin-firebase_app_distribution-0.3.7/lib/fastlane/plugin/firebase_app_distribution/client/firebase_app_distribution_api_client.rb:112:in `upload_binary'
        from /Library/Ruby/Gems/2.6.0/gems/fastlane-plugin-firebase_app_distribution-0.3.7/lib/fastlane/plugin/firebase_app_distribution/client/firebase_app_distribution_api_client.rb:142:in `upload'
        from /Library/Ruby/Gems/2.6.0/gems/fastlane-plugin-firebase_app_distribution-0.3.7/lib/fastlane/plugin/firebase_app_distribution/actions/firebase_app_distribution_action.rb:45:in `run'
        from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.210.1/fastlane/lib/fastlane/runner.rb:263:in `block (2 levels) in execute_action'
        from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.210.1/fastlane/lib/fastlane/actions/actions_helper.rb:69:in `execute_action'
        from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.210.1/fastlane/lib/fastlane/runner.rb:255:in `block in execute_action'
        from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.210.1/fastlane/lib/fastlane/runner.rb:229:in `chdir'
        from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.210.1/fastlane/lib/fastlane/runner.rb:229:in `execute_action'
        from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.210.1/fastlane/lib/fastlane/runner.rb:157:in `trigger_action_by_name'
        from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.210.1/fastlane/lib/fastlane/fast_file.rb:159:in `method_missing'
        from Fastfile:23:in `block (2 levels) in parsing_binding'
        from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.210.1/fastlane/lib/fastlane/lane.rb:33:in `call'
        from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.210.1/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
        from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.210.1/fastlane/lib/fastlane/runner.rb:45:in `chdir'
        from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.210.1/fastlane/lib/fastlane/runner.rb:45:in `execute'
        from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.210.1/fastlane/lib/fastlane/lane_manager.rb:47:in `cruise_lane'
        from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.210.1/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
        from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.210.1/fastlane/lib/fastlane/commands_generator.rb:110:in `block (2 levels) in run'
        from /Library/Ruby/Gems/2.6.0/gems/commander-4.6.0/lib/commander/command.rb:187:in `call'
        from /Library/Ruby/Gems/2.6.0/gems/commander-4.6.0/lib/commander/command.rb:157:in `run'
        from /Library/Ruby/Gems/2.6.0/gems/commander-4.6.0/lib/commander/runner.rb:444:in `run_active_command'
        from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.210.1/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:124:in `run!'
        from /Library/Ruby/Gems/2.6.0/gems/commander-4.6.0/lib/commander/delegates.rb:18:in `run!'
        from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.210.1/fastlane/lib/fastlane/commands_generator.rb:354:in `run'
        from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.210.1/fastlane/lib/fastlane/commands_generator.rb:43:in `start'
        from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.210.1/fastlane/lib/fastlane/cli_tools_distributor.rb:123:in `take_off'
        from /Library/Ruby/Gems/2.6.0/gems/fastlane-2.210.1/bin/fastlane:23:in `<top (required)>'
        from /usr/local/bin/fastlane:23:in `load'
        from /usr/local/bin/fastlane:23:in `<main>'
```

**Comment by iadcialim** [comment]
I have the same issue. Any solution to this?
